File: FrontEnd/stock-analyzer/src/app/types/BulkDeal.ts
export type BulkDeal = {
    clientName: string,
    dealDate: Date,
    companySymbol: string,
    companyFullName: string,
    stockAction: StockAction,
    quantity: Number,
    tradePrice: Number,
    remarks: string | null
}

export enum StockAction {
    Buy = 0,
    Sell = 1
}

/*companySymbol: string,
companyFullName: string,
date: Date,
series: string,
previousClose: Number,
openPrice: Number,
highPrice: Number,
lowPrice: Number,
lastPrice: Number,
closePrice: Number,
avgPrice: Number,
*/

File: Stock Analyzer/Stock Analyzer Domain/Models/BulkDeal.cs
namespace Stock_Analyzer_Domain.Models
{
  public class BulkDeal
  {
    public Guid Id { get; set; }
    public Client Client { get; set; }
    public DateTime DealDate { get; set; }
    public Company Company { get; set; }
    public string CompanyFullName { get; set; }
    public StockAction StockAction { get; set; }
    public long Quantity { get; set; }
    public double TradePrice { get; set; }
    public string? Remarks { get; set; }
  }
}
